### DISCUSSION TITLE
套了CDN后登录后台无法跳转

### DISCUSSION THREAD

**Discussion by yongzhi-xyz** [halo-dev/halo#5725] [Q&A]
### System information

- 外部访问地址: https://www.ruoyuit.com
- 启动时间: 2024-04-14 00:03
- 版本: 2.14.0
- 构建时间: 2024-04-01 11:15
- Git Commit: ae274db
- Java: IBM Semeru Runtime Open Edition / 21.0.2+13-LTS
- 数据库: MySQL Community Server - GPL / 8.2.0
- 操作系统: Linux / 5.4.119-20.0009.29
- 已激活主题: [Theme Joe3](https://github.com/jiewenhuang/halo-theme-joe3.0)
- 已启动插件:
  - [Willow Markdown 编辑器](https://github.com/guqing/willow-mde)
  - [ByteMD](https://github.com/ruibaby/plugin-bytemd)
  - [文章导入导出](https://jhacker.cn/2023/halo-plugin-export2doc)
  - [应用市场](https://github.com/halo-dev/plugin-app-store)
  - [Sitemap](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件](https://github.com/halo-dev/plugin-search-widget)
  - [RSS](https://github.com/halo-dev/plugin-feed)
  - [评论组件](https://github.com/halo-dev/plugin-comment-widget)

### What is the project operation method?

Docker Compose

### What happened?

1.套用CDN后登录后台，刚开始显示登录成功但是无法跳转到后台（显示登录成功后就没反应了），显示密码错误；现在是显示密码错误（没有输错）

2.使用IP+端口，登录可以成功进入后台

CDN缓存配置：
缓存键规则配置

| 类型   | 内容   | 忽略参数 | 忽略大小写 | 操作   |
| ---- | ---- | ---- | ----- | ---- |
| 全部文件 | 全部文件 | 不忽略  | 否     | 修改   |
| 文件后缀 | html | 全部忽略 | 否     | 修改删除 |

节点缓存过期配置

| 类型   | 内容                                                                                         | 缓存行为       | 优先级权重 | 操作   |
| ---- | ------------------------------------------------------------------------------------------ | ---------- | ----- | ---- |
| 全部文件 | 全部文件                                                                                       | 缓存1天，强制缓存  | 1     | 修改删除 |
| 文件目录 | /console                                                                                   | 不缓存        | 2     | 修改删除 |
| 文件后缀 | php;asp;aspx;jsp;cgi                                                                       | 不缓存        | 3     | 修改删除 |
| 文件后缀 | js;css;png;jpg;jpeg;gif;ico;bmp;swf;eot;svg;ttf;woff;woff2;webp;html;htm                   | 缓存30天，强制缓存 | 4     | 修改删除 |
| 文件后缀 | mp3;wma;flv;mp4;wmv;ogg;avi;m3u8;ts;doc;docx;xls;xlsx;ppt;pptx;txt;pdf;zip;exe;tat;swf;apk | 缓存30天，强制缓存 | 5     | 修改删除 |

OpenResty配置：
```
server {
    listen 80 ; 
    listen [::]:80 ; 
    listen 443 ssl http2 ; 
    listen [::]:443 ssl http2 ; 
    server_name www.ruoyuit.com; 
    index index.php index.html index.htm default.php default.htm default.html; 
    proxy_set_header Host $host; 
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 
    proxy_set_header X-Forwarded-Host $server_name; 
    proxy_set_header X-Real-IP $remote_addr; 
    proxy_http_version 1.1; 
    proxy_set_header Upgrade $http_upgrade; 
    proxy_set_header Connection "upgrade"; 
    access_log /www/sites/www.ruoyuit.com/log/access.log; 
    error_log /www/sites/www.ruoyuit.com/log/error.log; 
    access_by_lua_file /www/common/waf/access.lua; 
    set $RulePath /www/sites/www.ruoyuit.com/waf/rules; 
    set $logdir /www/sites/www.ruoyuit.com/log; 
    set $redirect on; 
    set $attackLog on; 
    set $CCDeny on; 
    set $urlWhiteAllow off; 
    set $urlBlockDeny off; 
    set $argsDeny on; 
    set $postDeny on; 
    set $cookieDeny on; 
    set $fileExtDeny on; 
    set $ipBlockDeny off; 
    set $ipWhiteAllow off; 
    location ^~ /.well-known/acme-challenge {
        allow all; 
        root /usr/share/nginx/html; 
    }
    location / {
        proxy_pass http://127.0.0.1:端口; 
    }
    if ($scheme = http) {
        return 301 https://$host$request_uri; 
    }
    ssl_certificate /www/sites/www.ruoyuit.com/ssl/fullchain.pem; 
    ssl_certificate_key /www/sites/www.ruoyuit.com/ssl/privkey.pem; 
    ssl_protocols TLSv1.3 TLSv1.2 TLSv1.1 TLSv1; 
    ssl_ciphers EECDH+CHACHA20:EECDH+CHACHA20-draft:EECDH+AES128:RSA+AES128:EECDH+AES256:RSA+AES256:EECDH+3DES:RSA+3DES:!MD5; 
    ssl_prefer_server_ciphers on; 
    ssl_session_cache shared:SSL:10m; 
    ssl_session_timeout 10m; 
    add_header Strict-Transport-Security "max-age=31536000"; 
    error_page 497 https://$host$request_uri; 
    proxy_set_header X-Forwarded-Proto https; 
    ssl_stapling on; 
    ssl_stapling_verify on; 
    location ~ .*\.(js|css|png|jpg|jpeg|gif|ico|bmp|swf|eot|svg|ttf|woff|woff2|webp)$ {
        log_not_found off; 
        valid_referers www.ruoyuit.com ruoyuit.com; 
        if ($invalid_referer) {
            return 404; 
            access_log off; 
        }
    }
    limit_conn perserver 300; 
    limit_conn perip 25; 
    limit_rate 512k; 
}
```

### Relevant log output

```shell
网站下载日志：
2024-04-15T00:03:28.864+08:00  INFO 7 --- [scheduling-1] r.h.a.s.a.login.impl.RsaKeyService       : Generating RSA keys...
2024-04-15T00:03:29.210+08:00  INFO 7 --- [scheduling-1] r.h.a.s.a.login.impl.RsaKeyService       : Generated RSA keys. Usage: 345 ms.
```

**Comment by ruibaby** [comment]
这看起来并不是 Halo 本身的问题，所以我将转移到 Discussion。

此外，你当前的网站无法正常访问，如果可以的话，可以临时开放注册，我可以注册之后上去看看请求结果。

**Reply by yongzhi-xyz** [reply]
您说我的首页无法访问吗，我换环境可以访问啊，就是CDN后台无法登录

**Comment by ghost** [comment]
我也遇到了这个问题，ip地址加端口可以访问后台但是cdn＋域名在登录的时候会一直重定向回登录页面，你现在解决了吗
